### PR TITLE
Add -D_FORTIFY_SOURCE=2 for fixing hardening-check

### PR DESCRIPTION
**Pull request by PokhodenkoSA**
See https://wiki.debian.org/HardeningWalkthrough#Testing_your_packages_after_conversion

### CODE REVIEW

**Comment by diptorupd** [comment]
Required security compiler’s flags:

`-Wformat -Wformat-security -fstack-protector-all -D_FORTIFY_SOURCE=2 -fpic -fPIC -O3`

Linker flags:

`-Wl,-z,noexecstack,-z,relro,-z,now`

Any recent GLIBC and stdlibc++ would work, I thought.

File: backends/CMakeLists.txt [hunk 1]
```diff
@@ -76,12 +76,10 @@ if(WIN32)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Qstd=c++17")
     set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -ggdb3 -DDEBUG ")
 elseif(UNIX)
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS}                                        \
-       -Wall -Wextra -Winit-self -Wuninitialized -Wmissing-declarations        \
-       -fdiagnostics-color=auto -O3                                            \
-    ")
+    set(SDL_FLAGS "-fstack-protector -fstack-protector-all -fpic -fPIC -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -fno-strict-overflow -fno-delete-null-pointer-checks")
```
Comment (diptorupd) [inline backends/CMakeLists.txt:79]: Can we reformat the flags like this to make them readable. Also do we not need `-O3` ?
```
    set(CMAKE_C_FLAGS "                                                        \
        ${CMAKE_C_FLAGS}                                                       \
       -Wall                                                                   \
       -Wextra                                                                 \
       -Winit-self                                                             \
       -Wuninitialized                                                         \
       -Wmissing-declarations                                                  \
       -fdiagnostics-color=auto -O3                                            \
    ")
    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Qstd=c++17")
    set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -ggdb3 -DDEBUG ")
elseif(UNIX)
    set(SDL_FLAGS "                                                            \
        -fstack-protector                                                      \
        -fstack-protector-all                                                  \
        -fpic                                                                  \
        -fPIC                                                                  \
        -D_FORTIFY_SOURCE=2                                                    \
        -Wformat                                                               \
        -Wformat-security                                                      \
        -fno-strict-overflow                                                   \
        -fno-delete-null-pointer-checks                                        \
        "
    )
    set(CMAKE_C_FLAGS "                                                        \
        ${CMAKE_C_FLAGS}                                                       \
        ${SDL_FLAGS}                                                           \
        -Wall                                                                  \
        -Wextra                                                                \
        -Winit-self                                                            \
        -Wuninitialized                                                        \
        -Wmissing-declarations                                                 \
        -fdiagnostics-color=auto                                               \
        -O3                                                                    \
        "
    )
    set(CMAKE_C_FLAGS_DEBUG "                                                  \
        ${CMAKE_C_FLAGS_DEBUG}                                                 \
         ${SDL_FLAGS}                                                          \
        -Wall                                                                  \
        -Wextra                                                                \
        -Winit-self                                                            \
        -Wuninitialized                                                        \
        -Wmissing-declarations                                                 \
        -fdiagnostics-color=auto                                               \
        -O0                                                                    \
        -ggdb3                                                                 \
        -DDEBUG                                                                \
        "
    )
```

File: backends/CMakeLists.txt [hunk 2]
```diff
@@ -76,12 +76,10 @@ if(WIN32)
...
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${SDL_FLAGS} -Wall -Wextra -Winit-self -Wuninitialized -Wmissing-declarations -fdiagnostics-color=auto")
     set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -ggdb3 -DDEBUG ")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17 -fsycl")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${SDL_FLAGS} -std=c++17 -fsycl")
```
Comment (diptorupd) [inline backends/CMakeLists.txt:82]: `-O3` needs to be added?

Comment (PokhodenkoSA) [reply backends/CMakeLists.txt:82]: -O3 is part of CMAKE_CXX_FLAGS_RELEASE.

Comment (diptorupd) [reply backends/CMakeLists.txt:82]: Of course.

File: backends/CMakeLists.txt (hunk 1 above)
Comment (PokhodenkoSA) [reply backends/CMakeLists.txt:79]: This style of formatting makes a lot spaces in commands. You can see all calls to compilers if use `make -j4 VERBOSE=1`.

Comment (PokhodenkoSA) [reply backends/CMakeLists.txt:79]: There will be a lot of spaces.

Comment (PokhodenkoSA) [reply backends/CMakeLists.txt:79]: I would propose to use [`add_compile_options()`](https://cmake.org/cmake/help/v3.18/command/add_compile_options.html)

Comment (diptorupd) [reply backends/CMakeLists.txt:79]: Ok, Then never mind.

**Review by diptorupd — APPROVED** [review]

**Pull request merged**